src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -621,6 +633,45 @@ private MsalAccessTokenCacheItem FilterByKeyId(MsalAccessTokenCacheItem item, Au
             return null;
         }
 
+        private static IEnumerable<MsalRefreshTokenCacheItem> FilterRtsByHomeAccountIdOrAssertion(
+            AuthenticationRequestParameters requestParams,
+            IEnumerable<MsalRefreshTokenCacheItem> rtCacheItems,
+            string familyId)
+        {
+            if (requestParams.ApiId == ApiEvent.ApiIds.AcquireTokenOnBehalfOf) // OBO
+            {
+                rtCacheItems = rtCacheItems.FilterWithLogging(item =>
+                                !string.IsNullOrEmpty(item.UserAssertionHash) &&
+                                item.UserAssertionHash.Equals(requestParams.UserAssertion.AssertionHash, StringComparison.OrdinalIgnoreCase),
+                                requestParams.RequestContext.Logger,
+                                "Filtering by user assertion id");
+            }
+            else
+            {
+                rtCacheItems = rtCacheItems.FilterWithLogging(item => item.HomeAccountId.Equals(
+                                requestParams.Account.HomeAccountId?.Identifier, StringComparison.OrdinalIgnoreCase),
+                                requestParams.RequestContext.Logger,
+                                "Filtering by home account id");
+            }
+
+            rtCacheItems = rtCacheItems.FilterWithLogging(item =>
```

Comment:
nit

```csharp 
bool filterByFamilyId = string.IsNullOrEmtpy(familyId);
if (filterByFamilyId) 
{
      // filter by family ID
}
else
{
    // filter by client id
}

Reply:
Appreciate how much more readable RT code filtering is! The fact that we should filter either by clientID or by familyID was very subtle / poorly done, this is much more clear.